### PR TITLE
feat: add script to automate tagging of repose for stage promotion

### PR DESCRIPTION
**Pull request by kgilmer**
CC @khos2ow & @SoumyaRanjanPatnaik for visiblity.  I'm running this script locally today to create tags for release on `testing` stage.

### CODE REVIEW

**Comment by kgilmer** [comment]
Script output for `r3_2-beta1` run:

```
unstable-debian-bookworm arc-icon-theme r3_2-beta1 DEFAULT
unstable-debian-bookworm ayu-theme r3_2-beta1 DEFAULT
unstable-debian-bookworm childe r3_2-beta1-debian-bookworm SPECIAL
unstable-debian-bookworm dracula-gtk r3_2-beta1 DEFAULT
unstable-debian-bookworm fonts-jetbrains-mono r3_2-beta1 DEFAULT
unstable-debian-bookworm fonts-materialdesignicons-webfont r3_2-beta1 DEFAULT
unstable-debian-bookworm fonts-nerd-fonts r3_2-beta1 DEFAULT
unstable-debian-bookworm fonts-source-code-pro-ttf r3_2-beta1 DEFAULT
unstable-debian-bookworm gruvbox-gtk r3_2-beta1 DEFAULT
unstable-debian-bookworm i3-next-workspace r3_2-beta1 DEFAULT
unstable-debian-bookworm i3-snapshot r3_2-beta1 DEFAULT
unstable-debian-bookworm i3-swap-focus r3_2-beta1 DEFAULT
unstable-debian-bookworm i3-wm r3_2-beta1 DEFAULT
unstable-debian-bookworm i3xrocks r3_2-beta1 DEFAULT
unstable-debian-bookworm ilia r3_2-beta1-ubuntu-jammy SPECIAL
unstable-debian-bookworm lago r3_2-beta1-ubuntu-jammy SPECIAL
unstable-debian-bookworm nordic r3_2-beta1 DEFAULT
unstable-debian-bookworm picom r3_2-beta1 DEFAULT
unstable-debian-bookworm python3-i3ipc r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-compositor-compton-glx r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-compositor-none r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-compositor-picom-glx r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-compositor-xcompmgr r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-control-center r3_2-beta1-gnome-43 SPECIAL
unstable-debian-bookworm regolith-default-settings r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-desktop r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-distro-ubuntu r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-ftue r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-i3xrocks-config r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-lightdm-config r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-look-default r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-look-extra r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-rofi-config r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-rofication r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-session r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-system-ubuntu r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-unclutter-xfixes r3_2-beta1 DEFAULT
unstable-debian-bookworm regolith-wm-config r3_2-beta1 DEFAULT
unstable-debian-bookworm remontoire r3_2-beta1 DEFAULT
unstable-debian-bookworm solarc-theme r3_2-beta1 DEFAULT
unstable-debian-bookworm whitesur-gtk-theme r3_2-beta1 DEFAULT
unstable-debian-bookworm xdg-desktop-portal-regolith r3_2-beta1 DEFAULT
unstable-debian-bookworm xrescat r3_2-beta1 DEFAULT
unstable-debian-bullseye ilia r3_2-beta1 DEFAULT
unstable-debian-bullseye lago r3_2-beta1 DEFAULT
unstable-debian-bullseye plymouth-theme-regolith r3_2-beta1 DEFAULT
unstable-debian-bullseye regolith-control-center r3_2-beta1-debian-bullseye SPECIAL
unstable-debian-bullseye regolith-desktop r3_2-beta1-debian-bullseye SPECIAL
unstable-debian-testing armesto r3_2-beta1 DEFAULT
unstable-debian-testing gtklock r3_2-beta1-ubuntu-jammy SPECIAL
unstable-debian-testing i3status-rs r3_2-beta1 DEFAULT
unstable-debian-testing libtrawldb r3_2-beta1 DEFAULT
unstable-debian-testing regolith-avizo r3_2-beta1-ubuntu-jammy SPECIAL
unstable-debian-testing regolith-control-center r3_2-beta1-gnome-46 SPECIAL
unstable-debian-testing regolith-displayd r3_2-beta1 DEFAULT
unstable-debian-testing regolith-inputd r3_2-beta1 DEFAULT
unstable-debian-testing regolith-powerd r3_2-beta1-debian-bookworm SPECIAL
unstable-debian-testing sway-audio-idle-inhibit r3_2-beta1-debian-testing SPECIAL
unstable-debian-testing sway-regolith r3_2-beta1-debian-testing SPECIAL
unstable-debian-testing trawl r3_2-beta1 DEFAULT
unstable-ubuntu-focal regolith-control-center r3_2-beta1 DEFAULT
unstable-ubuntu-focal regolith-desktop r3_2-beta1-ubuntu-focal SPECIAL
unstable-ubuntu-jammy i3status-rs r3_2-beta1-ubuntu-jammy SPECIAL
unstable-ubuntu-jammy regolith-control-center r3_2-beta1-ubuntu-jammy SPECIAL
unstable-ubuntu-jammy regolith-powerd r3_2-beta1 DEFAULT
unstable-ubuntu-jammy sway-audio-idle-inhibit r3_2-beta1 DEFAULT
unstable-ubuntu-jammy sway-regolith r3_2-beta1-ubuntu-jammy SPECIAL
unstable-ubuntu-jammy ubiquity-slideshow-regolith r3_2-beta1 DEFAULT
unstable-ubuntu-noble childe r3_2-beta1 DEFAULT
unstable-ubuntu-noble sway-regolith r3_2-beta1 DEFAULT
```

File: .github/scripts/tag-stage.sh [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  elif [[ "$PACKAGE_SOURCE_REF" == "packaging/v1.9-regolith" && "$PACKAGE_NAME" == "sway-regolith" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian" && "$PACKAGE_NAME" == "fonts-nerd-fonts" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  else
+    echo "### Ignoring unhandled variant: $PACKAGE_NAME on $STAGE-$DISTRO-$CODENAME with tag $PACKAGE_SOURCE_REF"
+  fi
+
+  popd > /dev/null
+}
```
Comment (SoumyaRanjanPatnaik) [inline .github/scripts/tag-stage.sh:101]: This seems like a hassle. Maybe we can eventually migrate to a consistent naming strategy for the entire project?

Comment (kgilmer) [reply .github/scripts/tag-stage.sh:101]: > ...consistent naming strategy for the entire project?

Yes, in fact this PR is the first iteration to achieve that goal.  It's been a tedious and error prone process to do each release, as I've had to go into the package model and fix things up from memory.  This script will allow the first part of automating this: cut tags for a release based upon package model.  For 3.2 I'll stick to this.  For the next release I'll look into automating mutating the next-stage package model w/ the data.  Once that's working then releases should be automated and consistent.

Comment (SoumyaRanjanPatnaik) [reply .github/scripts/tag-stage.sh:101]: This sounds reasonable.

**Review by SoumyaRanjanPatnaik — APPROVED** [review]

File: .github/scripts/tag-stage.sh (hunk 1 above)
Comment (kgilmer) [reply .github/scripts/tag-stage.sh]: I've done some more work to document conventions in the script and further categories branch mappings as either conventions or special cases.

**Comment by kgilmer** [comment]
Demo: https://github.com/regolith-linux/voulage/actions/runs/9331407838

**Pull request merged**